pilot/proxy/envoy/config.go

```diff
@@ -906,6 +910,82 @@ func buildEgressHTTPRoutes(mesh *proxyconfig.MeshConfig, node proxy.Node,
 	return httpConfigs.normalize()
 }
 
+// buildEgressTCPListeners builds a listener on 0.0.0.0 per each distinct port of all TCP egress
+// rules and a cluster per each TCP egress rule
+func buildEgressTCPListeners(mesh *proxyconfig.MeshConfig, node proxy.Node,
+	config model.IstioConfigStore) (Listeners, Clusters) {
+
+	tcpListeners := make(Listeners, 0)
+	tcpClusters := make(Clusters, 0)
+
+	if node.Type == proxy.Router {
+		// No egress rule support for Routers. As semantics are not clear.
+		return tcpListeners, tcpClusters
+	}
+
+	egressRules, errs := model.RejectConflictingEgressRules(config.EgressRules())
+
+	if errs != nil {
+		glog.Warningf("Rejected rules: %v", errs)
+	}
+
+	tcpRulesByPort := make(map[int][]*proxyconfig.EgressRule)
+
+	for _, rule := range egressRules {
+		for _, port := range rule.Ports {
+			protocol := model.Protocol(strings.ToUpper(port.Protocol))
+			if protocol != model.ProtocolTCP {
+				continue
+			}
+			intPort := int(port.Port)
+
+			if _, ok := tcpRulesByPort[intPort]; ok {
+				tcpRulesByPort[intPort] = append(tcpRulesByPort[intPort], rule)
+			} else {
+				tcpRulesByPort[intPort] = []*proxyconfig.EgressRule{rule}
+			}
```

Comment:
`append(list, item)` handles the case where `list` is `nil`, so you can remove this check.

Reply:
@ijsnellf Thanks, let me fix that!